backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/common/utils/ansible/AnsibleRunnerHttpClient.java

```diff
@@ -134,17 +128,9 @@ public int processEvents(String playUuid,
             BiConsumer<String, String> fn,
             String msg,
             Path logFile) {
-        List<String> sortedEvents = new ArrayList<>();
-        try {
-            sortedEvents = getSortedEvents(playUuid, lastEventId);
-        } catch (InterruptedException ex) {
-            throw new AnsibleRunnerCallException(
-                    "Failed to get list of events for play: %1$s",
-                    playUuid);
-        }
-
         String jobEvents = getJobEventsDir(playUuid);
-        for (String event : sortedEvents) {
+        String event = getNextEvent(playUuid, lastEventId);
+        while(event != null){
```

Comment:
how about
```java
             while(true){
            String event = getNextEvent(playUuid, lastEventId);
            if (event == null) {
                break;
            }
```
instead of having two places that call getNextEvent?